config/value.go

```diff
@@ -333,10 +333,41 @@ func derefType(t reflect.Type) reflect.Type {
 	return t
 }
 
+func convertPrimitiveValue(value interface{}, targetType reflect.Type) (interface{}, error) {
```

Comment:
This is a refactor since this method was previously common between primitive type conversion and populateStruct. We don't want to do additional type conversion.